src/QueryEditorForm.test.tsx

```diff
@@ -45,134 +47,151 @@ beforeEach(() => {
+        expect(onChange).toHaveBeenCalledWith({
+          ...q,
+          connectionArgs: { ...q.connectionArgs, database: 'foo' },
+        });
+        expect(onRunQuery).not.toHaveBeenCalled();
+      });
+
+      it('should request select tables and not execute the query', async () => {
+        const onChange = jest.fn();
+        const onRunQuery = jest.fn();
+        ds.postResource = jest.fn().mockResolvedValue(['foo']);
+        ds.getTables = jest.fn((query) =>
+          ds.postResource('tables', {
+            region: query.connectionArgs.region,
+            catalog: query.connectionArgs.catalog,
+            database: query.connectionArgs.database,
+          })
+        );
+        render(<QueryEditorForm {...props} onChange={onChange} onRunQuery={onRunQuery} />);
+
+        const selectEl = screen.getByLabelText('Table');
+        expect(selectEl).toBeInTheDocument();
+
+        await select(selectEl, 'foo', { container: document.body });
+
+        expect(ds.postResource).toHaveBeenCalledWith(
+          'tables',
+          expect.objectContaining({ region: defaultKey, catalog: defaultKey, database: defaultKey })
+        );
+        expect(onChange).toHaveBeenCalledWith({
+          ...q,
+          table: 'foo',
+        });
+        expect(onRunQuery).not.toHaveBeenCalled();
+      });
+
+      it('should request select columns and not execute the query', async () => {
+        const onChange = jest.fn();
+        const onRunQuery = jest.fn();
+        ds.postResource = jest.fn().mockResolvedValue(['columnName']);
+        ds.getColumns = jest.fn((query) =>
+          ds.postResource('columns', {
+            region: query.connectionArgs.region,
+            catalog: query.connectionArgs.catalog,
+            database: query.connectionArgs.database,
+            table: query.table,
+          })
+        );
+        render(
+          <QueryEditorForm
+            {...props}
+            onChange={onChange}
+            onRunQuery={onRunQuery}
+            query={{ ...props.query, table: 'tableName' }}
+          />
+        );
+
+        const selectEl = screen.getByLabelText('Column');
+        expect(selectEl).toBeInTheDocument();
+
+        await select(selectEl, 'columnName', { container: document.body });
+
+        expect(ds.postResource).toHaveBeenCalledWith(
+          'columns',
+          expect.objectContaining({ region: defaultKey, catalog: defaultKey, database: defaultKey, table: 'tableName' })
+        );
+        expect(onChange).toHaveBeenCalledWith({
+          ...q,
+          column: 'columnName',
+          table: 'tableName',
+        });
+        expect(onRunQuery).not.toHaveBeenCalled();
+      });
+
+      it('should display query options by default', async () => {
+        render(<QueryEditorForm {...props} />);
+        // @ts-ignore
+        const selectEl = screen.getByLabelText(config.featureToggles.awsDatasourcesNewFormStyling ? 'Format dataframes as': 'Format as');
```

Comment:
Will change `dataframes` to `data frames` when [this](https://github.com/grafana/grafana-aws-sdk-react/pull/67) is merged and released